azext_iot/tests/digitaltwins/test_dt_twin_unit.py

```diff
@@ -168,6 +169,10 @@ def test_query_twins_error(self, fixture_cmd, service_client_error):
                 show_cost=False,
                 resource_group_name=None
             )
+            # For some reason the new responses pkg doesnt fail on the first 500...
+            status_codes = [call.response.status_code for call in service_client_error.calls]
+            assert status_codes == [500, 200]
+            raise CLIError()
```

Comment:
Does the service_client_error fixture need to be constructed in a different way with the new version? It is suspicious to have to raise CLIError in the test to fulfill what the application logic is supposed to throw.

Reply:
This is for the test case that has the first request as a 500 (with a next link) and the second request as a 200 (from the next link). The result is that the command succeeds even though there is a 500.

I cannot test this in a real service...

Reply:
Raising a CLIError in the test code that is supposed to validate the application code raises CLIError is wrong and not testing what its supposed to.

Looking at the respective digitaltwins client operation, if the response status code is not 200 it should raise models.ErrorResponseException.

![image](https://user-images.githubusercontent.com/28658112/217661825-2bf58566-7daa-4151-aa74-1c95cabd15c6.png)

You don't need a real service to test this flow, and at a minimum, one can change the status code of the response in a breakpoint before the response is inspected in further logic.

Reply:
The issue is that the code skips over the 500 (since there is a next link) and thus the final response status code is 200. So the case where the first link is a 500 with a next link to a 200 doesn't throw an error but succeeds...

Example:
`pytest azext_iot/tests/digitaltwins/test_dt_twin_unit.py::TestTwinQueryTwins::test_query_twins_error[service_client_error2]`

pdb prints:
![image](https://user-images.githubusercontent.com/73560279/217665524-b53c7b1b-f953-458d-9fc4-1e75477b37db.png)

Reply:
But I do understand that raising a CLIError in this case is very weird since the code does not error. My other suggestion is to remove this test case.

Reply:
I don't see how the code skips over the 500, when a 500 will interrupt page accumulation that happens with `accumulate_result`.

When I run the same test (with responses 0.20.0)

Right before the first API call (notice I have 2 break points here, before and after the call)

![image](https://user-images.githubusercontent.com/28658112/217670322-97be1323-1172-4b54-9080-aab43ca6d310.png)


After getting the first response by the digitaltwins client
![image](https://user-images.githubusercontent.com/28658112/217669046-f3d6fa97-a4cd-4135-95f0-03826e916a87.png)

It's a 500 so raise models.ErrorResponseException ....I will continue the debugger from here to the next breakpoint
![image](https://user-images.githubusercontent.com/28658112/217669321-8edff4f7-1a3b-4285-b918-6c6e3c4d2e8f.png)

The next breakpoint is the one that wraps accumulate result and uses `handle_service_exception` to throw something derived from CLIError.
![image](https://user-images.githubusercontent.com/28658112/217670526-e17f1ed7-5751-4676-a131-810f913c0223.png)

Which passes the existing test definition.

![image](https://user-images.githubusercontent.com/28658112/217670883-99f5efec-eff1-41e8-b0ba-d17bd09110e3.png)